src/test/java/WebDriverScenarioWithTests.java

```diff
@@ -0,0 +1,132 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import ru.yandex.qatools.ashot.AShot;
+import ru.yandex.qatools.ashot.Screenshot;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategies;
+
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class WebDriverScenarioWithTests {
+
+    private WebDriver driver;
+    Logger logger = Logger.getGlobal();
+
+    @BeforeMethod(alwaysRun = true)
+    public void browsersSetup() {
+        System.setProperty("webdriver.chrome.driver", "src\\test\\resources\\webdrivers\\chromedriver.exe");
+        driver = new ChromeDriver();
+        System.setProperty("webdriver.edge.driver", "src\\test\\resources\\webdrivers\\msedgedriver.exe");
+        driver = new EdgeDriver();
+        System.setProperty("webdriver.gecko.driver", "src\\test\\resources\\webdrivers\\geckodriver.exe");
+        driver = new FirefoxDriver();
+    }
+
+    @Test
+    public void commonFlow() throws IOException {
+
+        ExpectedCondition<Boolean> pageLoadCondition = new
+                ExpectedCondition<Boolean>() {
+                    public Boolean apply(WebDriver driver) {
+                        return ((JavascriptExecutor) driver).executeScript("return document.readyState").equals("complete");
+                    }
+                };
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        logger.info("1. Launch Chrome browser in maximized mode.");
+        driver.get("https://google.com");
+        wait.until(pageLoadCondition);
+        driver.manage().window().maximize();
+
+        logger.info("2. Open 'google.com' and search for 'wikipedia'");
+        WebElement searchInput = driver.findElement(By.name("q"));
+        searchInput.sendKeys("wikipedia" + Keys.ENTER);
+        List<WebElement> searchResults = driver.findElements(By.xpath("//cite[contains(text())]"));
+        logger.info(String.format("Search results number for requested text: %s", searchResults.size()));
+        Assert.assertTrue(Boolean.parseBoolean(String.valueOf(searchResults.contains("wikipedia"))), "Search results do not contain expected text.");
```

Comment:
why not just `searchResults.contains("wikipedia")` ?)

Reply:
Probably this was too easy for me...))
Corrected, thank you.